pool/flexpool.go

```diff
@@ -0,0 +1,77 @@
+package pool
+
+import (
+	"sync"
+)
+
+type FlexPool struct {
+	mu      sync.Mutex
+	taskQueue map[string]chan func()
+	useCount map[string]int
+	terminating bool
+}
+
+func NewFlexPool() *FlexPool {
+	// taskQueue is unbuffered since items are always removed immediately.
```

Comment:
this is copying a comment from the fix pool that doesn't seem to be correct -- the individual queues are buffered and the tasks are only removed as they are completed